Packs/GenericAPIEventCollector/Integrations/GenericAPIEventCollector/GenericAPIEventCollector_test.py

```diff
@@ -0,0 +1,417 @@
+from datetime import datetime
+from unittest.mock import patch
+
+import pytest
+
+from CommonServerPython import DemistoException
+from GenericAPIEventCollector import (
+    datetime_to_timestamp_format, timestamp_format_to_datetime, recursive_replace,
+    get_time_field_from_event_to_dt, is_pagination_needed, iso8601_to_datetime_str, try_load_json,
+    parse_json_param, generate_authentication_headers,
+    extract_pagination_params, PaginationLogic, TimestampFieldConfig, organize_events_to_xsiam_format, setup_search_events,
+    RequestData, generate_headers
+)
+
+
+def test_datetime_to_timestamp_format():
+    dt = datetime(2023, 5, 1, 12, 0, 0)
+    assert datetime_to_timestamp_format(dt, '%Y-%m-%dT%H:%M:%SZ') == '2023-05-01T12:00:00Z'
+    assert datetime_to_timestamp_format(dt, 'epoch') == str(dt.timestamp())
+
+
+def test_timestamp_format_to_datetime():
+    dt_str = '2023-05-01T12:00:00Z'
+    assert timestamp_format_to_datetime(dt_str, '%Y-%m-%dT%H:%M:%SZ') == datetime(2023, 5, 1, 12, 0, 0)
+    epoch_str = str(datetime(2023, 5, 1, 12, 0, 0).timestamp())
+    assert timestamp_format_to_datetime(epoch_str, 'epoch') == datetime(2023, 5, 1, 12, 0, 0)
+
+
+def test_recursive_replace():
+    org_dict = {'key1': 'value1', 'key2': {'key3': 'value3'}}
+    substitutions = [('value1', 'new_value1'), ('value3', 'new_value3')]
+    result = recursive_replace(org_dict, substitutions)
+    assert result == {'key1': 'new_value1', 'key2': {'key3': 'new_value3'}}
+
+
+def test_recursive_replace_with_none():
+    assert recursive_replace(None, []) is None
+
+
+def test_get_time_field_from_event_to_dt():
+    event = {'timestamp': '2023-05-01T12:00:00Z'}
+    config = TimestampFieldConfig(['timestamp'], '%Y-%m-%dT%H:%M:%SZ')
+    result = get_time_field_from_event_to_dt(event, config)
+    assert isinstance(result, datetime)
+    assert result == datetime(2023, 5, 1, 12, 0, 0)
+
+
+def test_get_time_field_from_event_to_dt_throws_exception():
+    event = {'non_timestamp_field': '2023-05-01T12:00:00Z'}
+    config = TimestampFieldConfig(['timestamp'], '%Y-%m-%dT%H:%M:%SZ')
+    with pytest.raises(DemistoException) as exc_info:
+        get_time_field_from_event_to_dt(event, config)
+    assert "Timestamp field: ['timestamp'] not found in event" in str(exc_info.value)
+
+
+def test_is_pagination_needed():
+    events = {'next_page': 'page2', 'has_more': True}
+    pagination_logic = PaginationLogic(True, ['next_page'], ['has_more'])
+    needed, next_page = is_pagination_needed(events, pagination_logic)
+    assert needed is True
+    assert next_page == 'page2'
+
+
+def test_is_pagination_not_needed():
```

Comment:
In most cases when pagination is not needed, the next page will no be accounted for.